src/importlinter/domain/helpers.py

```diff
@@ -43,7 +52,7 @@ def pop_imports(
 
 
 def import_expressions_to_imports(
-    graph: ImportGraph, expressions: Iterable[ImportExpression]
+    graph: ImportGraph, expressions: Iterable[ImportExpression], if_not_matched: AlertingLevels
```

Comment:
So, if I understood correctly:
- instead of using `pop_import_expressions()` we implement and use a `resolve_import_expressions()`
- `resolve_import_expressions()` will return a tuple of resolved and unresolved imports 
- if the unresolved imports part of the return value of `resolve_import_expressions()` is not empty we decide how to notify the user (error, warning or none)
- we keep `pop_import_expressions()` around for anyone using it in their custom contracts

Is it correct?

Reply:
> resolve_import_expressions() will return a tuple of resolved and unresolved imports

Well, strictly speaking unresolved _import expressions_, but otherwise yes, all correct!

Reply:
Did the refactor at the best of my understanding of the changes. Let me know if they make sense.

Reply:
Thank you, looking forward to reviewing.